Builds failing with grub packaging error

https://github.com/elementary/os/runs/3226938334#step:4:14674

```
The following packages will be REMOVED:
  grub-efi-amd64-bin* grub-efi-amd64-signed* grub-efi-ia32-bin* mtools*
  shim-signed*
WARNING: The following essential packages will be removed.
This should NOT be done unless you know exactly what you are doing!
  shim-signed grub-efi-amd64-signed (due to shim-signed)
E: Essential packages were removed and -y was used without --allow-remove-essential.
0 upgraded, 0 newly installed, 5 to remove and 0 not upgraded.
P: Begin unmounting filesystems...
P: Saving caches...
Error: Process completed with exit code 100.
```

I can reproduce on a local build machine as well.

--- a/build.sh
+++ b/build.sh
@@ -30,6 +30,8 @@ apt-get install -y live-build patch ubuntu-keyring
 # https://salsa.debian.org/live-team/live-build/merge_requests/31
 patch -d /usr/lib/live/build/ < live-build-fix-syslinux.patch
 
+patch -d /usr/lib/live/build/ < live-build-fix-shim-remove.patch
+
 # TODO: Remove this once debootstrap 1.0.117 or newer is released and available:
 # https://salsa.debian.org/installer-team/debootstrap/blob/master/debian/changelog
 ln -sfn /usr/share/debootstrap/scripts/gutsy /usr/share/debootstrap/scripts/focal

--- a/live-build-fix-shim-remove.patch
+++ b/live-build-fix-shim-remove.patch
@@ -1,0 +1,18 @@
+--- /usr/lib/live/build/binary_grub-efi	2019-03-11 10:05:40.000000000 +0000
++++ /usr/lib/live/build/binary_grub-efi_v2	2021-08-04 13:37:20.064547041 +0000
+@@ -267,8 +267,12 @@
+ 		# Saving cache
+ 		Save_cache cache/packages.binary
+
+-		# Removing depends
+-		Remove_package
++		# Removing depends, some bootloader packages are marked as Protected/Important
++		# in Ubuntu, so temporarily add an apt flag to allow them to be removed
++		PRE_APT_OPTIONS="${APT_OPTIONS}"
++		APT_OPTIONS="${APT_OPTIONS} --allow-remove-essential"
++		Remove_package
++		APT_OPTIONS="${PRE_APT_OPTIONS}"
+ 		;;
+
+ 	false)
+

